### PR TITLE
DIRECTOR: Implement erase lingo command

### PR DESCRIPTION
**Pull request by r41k0u**
Implements the erase LINGO command as `b_erase()` in lingo_builtins.

### CODE REVIEW

File: engines/director/lingo/lingo-builtins.cpp [hunk 1]
```diff
@@ -1737,9 +1737,18 @@ void LB::b_editableText(int nargs) {
...
 void LB::b_erase(int nargs) {
-	g_lingo->printSTUBWithArglist("b_erase", nargs);
+	Datum d = g_lingo->pop();
+	CastMember *eraseCast = g_director->getCurrentMovie()->getCastMember(d.asMemberID());
 
-	g_lingo->dropStack(nargs);
+	if (eraseCast) {
+		eraseCast->_erase = true;
+
+		for (uint i = 0; i < g_director->getCurrentMovie()->getScore()->_channels.size(); i++) {
```
Comment (mgerhardy) [inline engines/director/lingo/lingo-builtins.cpp:1746]: maybe the `g_director->getCurrentMovie()->getScore()` or even `g_director->getCurrentMovie()->getScore()->_channels` should be in a local variable as pointer or reference to reduce code duplication here.

File: engines/director/score.cpp [hunk 2]
```diff
@@ -556,9 +556,22 @@ void Score::renderSprites(uint16 frameId, RenderMode mode) {
 		}
 
 		if (channel->isDirty(nextSprite) || widgetRedrawn || mode == kRenderForceUpdate) {
-			if (!currentSprite->_trails)
+			if (currentSprite && !currentSprite->_trails)
 				_window->addDirtyRect(channel->getBbox());
 
+			if (currentSprite->_cast && currentSprite->_cast->_erase) {
+				for (Common::HashMap<int, CastMember *>::iterator it = currentSprite->_cast->getCast()->_loadedCast->begin(); it != currentSprite->_cast->getCast()->_loadedCast->end(); ++it) {
```
Comment (sev-) [inline engines/director/score.cpp:563]: Write
```C++
auto cast = currentSprite->_cast->getCast()->_loadedCast;
for (auto it = cast->begin(); it != casr->end(); ++it) {
    ...
    cast->erase(it);
}
```

**Comment by sev-** [comment]
Thank you! Now it looks proper. Though, in the future, I recommend use `auto` more often. Since we wrote Director code, we allowed C++11 usage.

**Comment by Gauravmauryaios** [comment]
`auto cast = currentSprite->_cast->getCast()->_loadedCast;
for (auto it = cast->begin(); it != casr->end(); ++it) {
    ...
    cast->erase(it);
    break;
}`

**Comment by sev-** [comment]
> `auto cast = currentSprite->_cast->getCast()->_loadedCast; for (auto it = cast->begin(); it != casr->end(); ++it) { ... cast->erase(it); break; }`

What is your point here?

**Comment by Gauravmauryaios** [comment]
Implements the erase LINGO @sev-

**Pull request merged**